src/components/ec-phone-number-input/ec-phone-number-input.vue

```diff
@@ -149,197 +149,186 @@
-        });
-      },
-    },
-    selectedCountryAreaCode() {
-      return this.modelValue?.country?.areaCode || null;
-    },
-    selectedCountryImage() {
-      return this.getCountryFlagPath(this.modelValue?.country?.countryCode);
-    },
-    selectedCountryName() {
-      return this.modelValue?.country?.name;
-    },
-    isInvalid() {
-      return !!this.errorMessage;
-    },
+  bottomNote: {
+    type: String,
   },
-  methods: {
-    getPopoverStyle() {
-      if (this.$refs.popoverWidthReference) {
-        return {
-          width: `${this.$refs.popoverWidthReference.offsetWidth}px`,
-        };
-      }
-
-      return null;
-    },
-    onFocusCountry() {
-      this.countriesHasFocus = true;
-      this.$emit('focus');
-    },
-    onCountryChange(evt) {
-      this.countriesHasFocus = true;
-      this.$emit('change', evt);
-      this.$emit('country-change', evt);
-    },
-    onPhoneNumberChange(evt) {
-      this.$emit('change', evt);
-      this.$emit('phone-number-change', evt);
-    },
-    getCountryFlagPath(countryCode) {
-      if (!countryCode) {
-        return null;
-      }
-      try {
-        // eslint-disable-next-line global-require, import/no-dynamic-require
-        return require(`svg-country-flags/png100px/${countryCode.toLowerCase()}.png`);
-      } catch (err) {
-        return null;
-      }
+  isWarning: {
+    type: Boolean,
+    default: false,
+  },
+  isMasked: {
+    type: Boolean,
+    default: false,
+  },
+  warningTooltipMessage: {
+    type: String,
+  },
+  errorMessage: {
+    type: String,
+  },
+  errorTooltipMessage: {
+    type: String,
+  },
+  countries: {
+    type: Array,
+  },
+  areCountriesLoading: {
+    type: Boolean,
+    default: false,
+  },
+  isDisabled: {
+    type: Boolean,
+    default: false,
+  },
+  disabledTooltipMessage: {
+    type: String,
+  },
+  isSensitive: {
+    type: Boolean,
+    default: false,
+  },
+  countryPlaceholder: {
+    type: String,
+  },
+  phoneNumberPlaceholder: {
+    type: String,
+  },
+  isSearchEnabled: {
+    type: Boolean,
+    default: true,
+  },
+  searchCountryPlaceholder: {
+    type: String,
+    default: 'Search...',
+  },
+  noCountriesText: {
+    type: String,
+    default: 'No results found',
+  },
+  level: {
+    type: String,
+    validator(value) {
+      return ['notification', 'modal', 'tooltip', 'level-1', 'level-2', 'level-3'].includes(value);
     },
   },
-};
+});
+
+const emit = defineEmits([
+  'update:modelValue',
+  'change',
+  'focus',
+  'open',
+  'after-open',
+  'country-change',
+  'phone-number-change',
+]);
+
+const uid = getUid();
+const countriesHasFocus = ref(false);
+const popoverOptions = ref({
+  autoSize: 'min',
+});
+const popoverWidthReference = ref(null);
+
+const id = computed(() => `ec-phone-number-input-field-${uid}`);
+const errorId = computed(() => (isInvalid.value ? `ec-phone-number-input-${uid}` : null));
+const countriesItems = computed(() => props.countries.map(country => ({
```

Comment:
"props.countries" maybe undefined and code will throw an error.

Reply:
corrected, inserted a default value